tools/oeedger8r/Emitter.ml

```diff
@@ -228,6 +175,51 @@ let oe_get_param_size (ptype, decl, argstruct) =
       else pa_size
   | _ -> ""
 
+(** TODO: Refactor!  *)
+let oe_get_param_count (ptype, decl, argstruct) =
+  (* Get the base type of the parameter, that is, recursively
+     decompose the pointer. *)
+  let atype =
+    match get_param_atype ptype with
+    | Ptr at -> at
+    | _ -> get_param_atype ptype
+  in
+  let base_t = get_tystr atype in
+  let type_expr =
+    match ptype with
+    | PTPtr (_, ptr_attr) ->
+        if ptr_attr.pa_isptr then sprintf "*(%s)0" base_t else base_t
+    | _ -> base_t
+  in
+  (* Convert an attribute to string. *)
+  let attr_value_to_string av =
+    match av with
+    | None -> "1"
+    | Some (ANumber n) -> string_of_int n
+    | Some (AString s) -> sprintf "%s%s" argstruct s
+    (* another parameter name *)
+  in
+  let pa_size_to_string pa =
+    let c = attr_value_to_string pa.ps_count in
+    if c <> "" then sprintf "%s" c
+    else
+      sprintf "(%s / sizeof(%s))" (attr_value_to_string pa.ps_size) type_expr
+  in
+  let decl_size_to_string (ptype : parameter_type) (d : declarator) =
+    let dims = List.map string_of_int d.array_dims in
+    String.concat " * " dims
+  in
+  match ptype with
+  | PTPtr (_, ptr_attr) ->
+      let pa_size = pa_size_to_string ptr_attr.pa_size in
+      (* Compute declared size *)
+      let decl_size = decl_size_to_string ptype decl in
+      if ptr_attr.pa_isstr || ptr_attr.pa_iswstr then "1"
```

Comment:
Instead of "1" we could use a string option or a new variant type that is more descriptive.